STAR.py

```diff
@@ -1,29 +1,60 @@
+from ast import Starred
 import pandas as pd
 import numpy as np
-
+import copy
+from typing import Union
 class InvalidElection(Exception):
     pass
 
 class TrueTie:
     def __init__(self, tied):
         self.tied = tied
-        self.elected = []
-
-def summary_data(ballots: pd.DataFrame):
+        self.elected = [] 
+class SummaryData:
+    def __init__(self,score_hist, score_sums, pairwise_matrix, preference_matrix):
+        self.score_hist = score_hist
+        self.score_sums = score_sums
+        self.pairwise_matrix = pairwise_matrix
+        self.preference_matrix = preference_matrix
+
+    def drop(self,candidates):
+        self.score_hist.drop(candidates, axis=0, inplace=True)
+        self.score_sums.drop(candidates, axis=0, inplace=True)
+        self.pairwise_matrix.drop(candidates, axis=0, inplace=True)
+        self.pairwise_matrix.drop(candidates, axis=1, inplace=True)
+        self.preference_matrix.drop(candidates, axis=0, inplace=True)
+        self.preference_matrix.drop(candidates, axis=1, inplace=True)
+        return self
+
+    def keep(self,candidates):
+        self.score_hist = self.score_hist.loc[candidates, :]
+        self.score_sums = self.score_sums.loc[candidates]
+        self.pairwise_matrix = self.pairwise_matrix.loc[candidates, candidates]
+        self.preference_matrix = self.preference_matrix.loc[candidates, candidates]
+        return self
+
+def get_summary_data(ballots: pd.DataFrame):
     # The below quantities can be emitted however is most useful
-    score_hist = {a : ballots[a].value_counts() for a in ballots.columns}
+    score_hist = pd.DataFrame(0, columns = range(6), index = ballots.columns)
```

Comment:
I know we're nearly always using max stars = 5, but magic numbers just make me uneasy. maybe we can parameterize this?